File: src/__fixtures__/oversized/distractors.ts
// Diverse small utilities (TASK-030 distractor corpus).
//
// Normal-sized single-chunk functions providing realistic top-5 competition,
// including topical neighbours of the colour-conversion and string tails so a
// truncated head embedding does not falsely rank for a tail query. Not A/B targets.

export function parseHexColor(hex: string): { r: number; g: number; b: number } | null {
  // Parse a #RRGGBB / #RGB hex colour string into 8-bit RGB channels.
  const cleaned = hex.trim().replace(/^#/, '');
  const expanded = cleaned.length === 3 ? cleaned.split('').map((c) => c + c).join('') : cleaned;
  if (!/^[0-9a-fA-F]{6}$/.test(expanded)) return null;
  return {
    r: parseInt(expanded.slice(0, 2), 16),
    g: parseInt(expanded.slice(2, 4), 16),
    b: parseInt(expanded.slice(4, 6), 16),
  };
}

export function relativeLuminance(r: number, g: number, b: number): number {
  // WCAG relative luminance of an sRGB colour, channels in 0..255.
  const channel = (v: number): number => {
    const s = v / 255;
    return s <= 0.03928 ? s / 12.92 : Math.pow((s + 0.055) / 1.055, 2.4);
  };
  return 0.2126 * channel(r) + 0.7152 * channel(g) + 0.0722 * channel(b);
}

export function clamp(value: number, lo: number, hi: number): number {
  // Constrain a value to the inclusive [lo, hi] range.
  return Math.max(lo, Math.min(hi, value));
}

export function slugify(text: string): string {
  // Turn arbitrary text into a url-safe lowercase slug.
  return text
    .toLowerCase()
    .trim()
    .replace(/[^a-z0-9]+/g, '-')
    .replace(/^-+|-+$/g, '');
}

export function chunkArray<T>(items: T[], size: number): T[][] {
  // Split an array into consecutive chunks of at most `size` elements.
  if (size <= 0) throw new Error('chunk size must be positive');
  const out: T[][] = [];
  for (let i = 0; i < items.length; i += size) {
    out.push(items.slice(i, i + size));
  }
  return out;
}

export function groupBy<T, K extends string>(items: T[], keyFn: (item: T) => K): Record<K, T[]> {
  // Group array elements into a record keyed by a derived key.
  const out = {} as Record<K, T[]>;
  for (const item of items) {
    const key = keyFn(item);
    (out[key] ??= []).push(item);
  }
  return out;
}

File: src/__fixtures__/oversized/distractors.py
"""Diverse small utilities (TASK-030 distractor corpus).

These are normal-sized single-chunk functions. They exist to give the A/B corpus
realistic top-5 competition — including a few *topical neighbours* of each tail
concept (networking/retry, string distance) so a truncated head embedding does
not falsely rank for a tail query. None of them is an A/B target.
"""

from typing import Callable, Iterable


def http_status_category(code: int) -> str:
    """Classify an HTTP status code into a coarse category."""
    if 100 <= code < 200:
        return "informational"
    if 200 <= code < 300:
        return "success"
    if 300 <= code < 400:
        return "redirect"
    if 400 <= code < 500:
        return "client-error"
    return "server-error"


def token_bucket_allow(tokens: float, capacity: float, refill_rate: float, elapsed: float) -> bool:
    """Rate-limit decision: does the bucket have a token after refilling?"""
    refilled = min(capacity, tokens + refill_rate * elapsed)
    return refilled >= 1.0


def parse_retry_after(header: str) -> int:
    """Parse a Retry-After header value (delta-seconds) into an int, clamped >= 0."""
    try:
        return max(0, int(header.strip()))
    except (ValueError, AttributeError):
        return 0


def hamming_distance(a: str, b: str) -> int:
    """Count positions at which two equal-length strings differ."""
    if len(a) != len(b):
        raise ValueError("hamming_distance requires equal-length strings")
    return sum(1 for x, y in zip(a, b) if x != y)


def longest_common_prefix(strings: Iterable[str]) -> str:
    """Return the longest string prefix shared by every input string."""
    items = list(strings)
    if not items:
        return ""
    prefix = items[0]
    for s in items[1:]:
        while not s.startswith(prefix):
            prefix = prefix[:-1]
            if prefix == "":
                return ""
    return prefix


def format_bytes(count: int) -> str:
    """Render a byte count in human-readable binary units."""
    size = float(count)
    for unit in ("B", "KiB", "MiB", "GiB", "TiB"):
        if size < 1024.0:
            return f"{size:.1f}{unit}"
        size /= 1024.0
    return f"{size:.1f}PiB"


def triangle_area(ax: float, ay: float, bx: float, by: float, cx: float, cy: float) -> float:
    """Area of a triangle from its three vertex coordinates (cross product)."""
    return abs((bx - ax) * (cy - ay) - (cx - ax) * (by - ay)) / 2.0


def bounding_box(points: list[tuple[float, float]]) -> tuple[float, float, float, float]:
    """Axis-aligned bounding rectangle (min_x, min_y, max_x, max_y) of 2D points."""
    xs = [x for x, _ in points]
    ys = [y for _, y in points]
    return (min(xs), min(ys), max(xs), max(ys))


def partition(items: Iterable[int], predicate: Callable[[int], bool]) -> tuple[list[int], list[int]]:
    """Split items into (matching, not-matching) by a predicate."""
    yes: list[int] = []
    no: list[int] = []
    for item in items:
        (yes if predicate(item) else no).append(item)
    return yes, no
